common/core/gpudevice.cpp

```diff
@@ -249,30 +249,28 @@ bool GpuDevice::DisplayManager::UpdateDisplayState() {
     // Ensure we have atleast one valid mode.
     if (connector->count_modes == 0)
       continue;
-
-    drmModeModeInfo mode;
-    memset(&mode, 0, sizeof(mode));
+    std::vector<drmModeModeInfo> mode;
+    uint32_t preferred_mode;
     bool found_prefered_mode = false;
     for (int32_t i = 0; i < connector->count_modes; ++i) {
-      mode = connector->modes[i];
+      mode.emplace_back(connector->modes[i]);
       // There is only one preferred mode per connector.
-      if (mode.type & DRM_MODE_TYPE_PREFERRED) {
-        found_prefered_mode = true;
-        break;
+      if (mode[i].type & DRM_MODE_TYPE_PREFERRED) {
+        preferred_mode = i;
       }
     }
 
-    if (!found_prefered_mode)
-      continue;
-
     // Lets try to find crts for any connected encoder.
     if (connector->encoder_id) {
       ScopedDrmEncoderPtr encoder(
           drmModeGetEncoder(fd_, connector->encoder_id));
       if (encoder && encoder->crtc_id) {
         for (auto &display : displays_) {
+          //Set the modes supported for each display
+          display->SetDrmModeInfo(mode);
```

Comment:
Instead of passing each one, we could just pass it to display once we have the full list created ?